Sources/Bot/Services/MergeService.swift

```diff
@@ -402,6 +412,31 @@ extension MergeService {
                 .map(Event.pullRequestDidChange)
         }
     }
+
+    /// Returns the consolidated status of all _required_ checks only
+    /// i.e. returns .failure or .pending if one of the required check is in .failure or .pending respectively
+    /// and return .success only if all required states are .success
+    fileprivate static func getRequiredChecksState(
+        github: GitHubAPIProtocol,
+        targetBranch: PullRequest.Branch,
+        commitState: CommitState
+    ) -> SignalProducer<CommitState.State, AnyError> {
+        if commitState.state == .success {
+            return .value(.success)
+        }
```

Comment:
this is because the `CommitState.state` property is already `.success` only if all checks (required or otherwise) are `.success`. So if all checks (required or otherwise) are successful, no need to make the API call to fetch the list of required checks on the target branch